awx/main/isolated/manager.py

```diff
@@ -40,6 +42,51 @@ def __init__(self, cancelled_callback=None, check_callback=None):
         self.idle_timeout = max(60, 2 * settings.AWX_ISOLATED_CONNECTION_TIMEOUT)
         self.started_at = None
         self.captured_command_artifact = False
+        self.instance = None
+
+    def build_inventory(self, hosts):
+        if self.instance and self.instance.is_containerized:
+            inventory = { 'all': { 'hosts': {} } }
+            for host in hosts:
+                inventory['all']['hosts'][host] = self.kubectl_or_oc_vars
+        else:
+            inventory = '\n'.join([
+                '{} ansible_ssh_user={}'.format(host, settings.AWX_ISOLATED_USERNAME)
+                for host in hosts
+            ])
+
+        return inventory
+
+    @property
+    def kubectl_or_oc_vars(self):
+        extravars = {}
+
+        credential = self.instance.instance_group.credential
+
+        extravars['ansible_connection'] = 'kubectl'
+        extravars['ansible_kubectl_host'] = credential.get_input('host')
+        extravars['ansible_kubectl_namespace'] = 'awx'
+
+        if credential.kind == 'kubernetes_bearer_token':
+            extravars['ansible_kubectl_token'] = credential.get_input('bearer_token')
+
+        if credential.get_input('verify_ssl'):
+            ca_cert_data = credential.get_input('ssl_ca_cert')
+            # TODO: Clean up this file after we use it.
+            with tempfile.NamedTemporaryFile(delete=False) as temp:
```

Comment:
We have a dir that all tmp files for this job should be created. We take care to cleanup these dirs; outside of this task in case of a bad failure.
 https://github.com/ansible/awx/blob/37119a453857b832b3cc9304b8c0f26e4d98e0bc/awx/main/isolated/manager.py#L141

Reply:
I fixed this up to use `self.private_data_dir`, I think that's what you were talking about? Resolving this convo.